File: fix_enc.js
const fs = require('fs');

const files = [
    "dashboard-rendimento.html",
    "selecionar-treinador.html",
    "turmas.html"
];

const replacements = [
    { p: "Gr?fico", r: "Gráfico" },
    { p: "M?dia", r: "Média" },
    { p: "Evolu??o", r: "Evolução" },
    { p: "Avalia??o", r: "Avaliação" },
    { p: "Di?rio", r: "Diário" },
    { p: "di?rio", r: "diário" },
    { p: "F?sico", r: "Físico" },
    { p: "T?cnico", r: "Técnico" },
    { p: "T?tico", r: "Tático" },
    { p: "S?rie", r: "Série" },
    { p: "s?rie", r: "série" },
    { p: "Frequ?ncia", r: "Frequência" },
    { p: "Card?aca", r: "Cardíaca" },
    { p: "M?xima", r: "Máxima" },
    { p: "Correla??o", r: "Correlação" },
    { p: "Les?o", r: "Lesão" },
    { p: "Les?es", r: "Lesões" },
    { p: "S?bado", r: "Sábado" },
    { p: "Lan?ar", r: "Lançar" },
    { p: "Competi??o", r: "Competição" },
    { p: "Aten??o", r: "Atenção" },
    { p: "Cl?nico", r: "Clínico" },
    { p: "S?b", r: "Sáb" },
    { p: "Vis??o", r: "Visão" },
    { p: "N?vel", r: "Nível" },
    { p: "Voc?", r: "Você" },
    { p: "?ltima", r: "Última" },
    { p: "?ltimo", r: "Último" },
    { p: "P?blico", r: "Público" },
    { p: "A?es", r: "Ações" },
    { p: "m?dia", r: "média" },
    { p: "Alcan?ado", r: "Alcançado" },
    { p: "Composi??o", r: "Composição" },
    { p: "Circunfer?ncia", r: "Circunferência" },
    { p: "Abd?men", r: "Abdômen" },
    { p: "Bra?o", r: "Braço" },
    { p: "Cut?nea", r: "Cutânea" },
    { p: "Tr?ceps", r: "Tríceps" },
    { p: "Regi?es", r: "Regiões" },
    { p: "Ocorr?ncia", r: "Ocorrência" },
    { p: "Incid?ncia", r: "Incidência" },
    { p: "M?s", r: "Mês" },
    { p: "Vit?ria", r: "Vitória" },
    { p: "Per?odo", r: "Período" },
    { p: "In?cio", r: "Início" },
    { p: "Pr?xima", r: "Próxima" },
    { p: "Transi??o", r: "Transição" },
    { p: "Equil?brio", r: "Equilíbrio" },
    { p: "Pontua??o", r: "Pontuação" },
    { p: "Val?ncia", r: "Valência" },
    { p: "Pot?ncia", r: "Potência" },
    { p: "Resist?ncia", r: "Resistência" },
    { p: "Aer?bia", r: "Aeróbia" },
    { p: "For?a", r: "Força" },
    { p: "Ac?mulo", r: "Acúmulo" },
    { p: "Sess?es", r: "Sessões" },
    { p: "Dura??o", r: "Duração" },
    { p: "Prontid?o", r: "Prontidão" },
    { p: "Distribui??o", r: "Distribuição" },
    { p: "Psicol?gico", r: "Psicológico" },
    { p: "Visualiza??o", r: "Visualização" },
    { p: "Relat?rio", r: "Relatório" },
    { p: "Ades?o", r: "Adesão" },
    { p: "Avan?ar", r: "Avançar" },
    { p: "Come?ar", r: "Começar" }
];

files.forEach(filepath => {
    if (fs.existsSync(filepath)) {
        try {
            let content = fs.readFileSync(filepath, 'utf-8');
            // string.replaceAll escapes literals safely natively
            replacements.forEach(({ p, r }) => {
                content = content.replaceAll(p, r);
            });
            fs.writeFileSync(filepath, content, 'utf-8');
            console.log(`Fixed ${filepath}`);
        } catch (e) {
            console.error(`Error on ${filepath}:`, e);
        }
    }
});

File: fix_enc.py
import os
import glob
import re

files = [
    "dashboard-rendimento.html",
    "selecionar-treinador.html",
    "turmas.html"
]

replacements = {
    r"M\ufffd?dia": "Média",
    r"m\ufffd?dia": "média",
    r"Evolu\ufffd?\ufffd?o": "Evolução",
    r"Avalia\ufffd?\ufffd?o": "Avaliação",
    r"A\ufffd?\ufffd?es": "Ações",
    r"T\ufffd?tico": "Tático",
    r"T\ufffd?cnico": "Técnico",
    r"Gr\ufffd?fico": "Gráfico",
    r"P\ufffd?blico": "Público",
    r"Di\ufffd?rio": "Diário",
    r"di\ufffd?rio": "diário",
    r"S\ufffd?bado": "Sábado",
    r"M\ufffd?dico": "Médico",
    r"Aten\ufffd?\ufffd?o": "Atenção",
    r"Cl\ufffd?nico": "Clínico",
    r"F\ufffd?sico": "Físico",
    r"Lan\ufffd?ar": "Lançar",
    r"S\ufffd?rie": "Série",
    r"s\ufffd?rie": "série",
    r"Frequ\ufffd?ncia": "Frequência",
    r"Card\ufffd?aca": "Cardíaca",
    r"M\ufffd?xima": "Máxima",
    r"Alcan\ufffd?ado": "Alcançado",
    r"Correla\ufffd?\ufffd?o": "Correlação",
    r"Composi\ufffd?\ufffd?o": "Composição",
    r"Circunfer\ufffd?ncia": "Circunferência",
    r"Abd(\ufffd?)men": "Abdômen",
    r"Bra\ufffd?o": "Braço",
    r"Cut\ufffd?nea": "Cutânea",
    r"Tr\ufffd?ceps": "Tríceps",
    r"Les\ufffd?es": "Lesões",
    r"Regi\ufffd?es": "Regiões",
    r"Ocorr\ufffd?ncia": "Ocorrência",
    r"Incid\ufffd?ncia": "Incidência",
    r"M\ufffd?s": "Mês",
    r"Vit\ufffd?ria": "Vitória",
    r"Per?odo": "Período",
    r"In\ufffd?cio": "Início",
    r"Pr\ufffd?xima": "Próxima",
    r"Transi\ufffd?\ufffd?o": "Transição",
    r"M\ufffd?s": "Mês",
    r"\?ltima": "Última",
    r"\?ltimo": "Último",
    r"Equil\ufffd?brio": "Equilíbrio",
    r"Pontua\ufffd?\ufffd?o": "Pontuação",
    r"Val\ufffd?ncia": "Valência",
    r"Pot\ufffd?ncia": "Potência",
    r"Resist\ufffd?ncia": "Resistência",
    r"Aer\ufffd?bia": "Aeróbia",
    r"For\ufffd?a": "Força",
    r"Ac\ufffd?mulo": "Acúmulo",
    r"Sess\ufffd?es": "Sessões",
    r"Dura\ufffd?\ufffd?o": "Duração",
    r"Prontid\ufffd?o": "Prontidão",
    r"Distribui\ufffd?\ufffd?o": "Distribuição",
    r"Psicol\ufffd?gico": "Psicológico",
    r"Visualiza\ufffd?\ufffd?o": "Visualização",
    r"Relat\ufffd?rio": "Relatório",
    r"Per?odo": "Período",
    r"Ades\ufffd?o": "Adesão",
}

for filepath in files:
    if os.path.exists(filepath):
        try:
            with open(filepath, 'r', encoding='utf-8', errors='replace') as f:
                content = f.read()

            for pattern, replacement in replacements.items():
                content = re.sub(pattern, replacement, content, flags=re.IGNORECASE)

            # Manual ? replacements where FFFD was not used
            content = content.replace("Gr?fico", "Gráfico")
            content = content.replace("M?dia", "Média")
            content = content.replace("Evolu??o", "Evolução")
            content = content.replace("Avalia??o", "Avaliação")
            content = content.replace("Di?rio", "Diário")
            content = content.replace("F?sico", "Físico")
            content = content.replace("T?cnico", "Técnico")
            content = content.replace("T?tico", "Tático")
            content = content.replace("S?rie", "Série")
            content = content.replace("Frequ?ncia", "Frequência")
            content = content.replace("Card?aca", "Cardíaca")
            content = content.replace("M?xima", "Máxima")
            content = content.replace("Correla??o", "Correlação")
            content = content.replace("Les?o", "Lesão")
            content = content.replace("Les?es", "Lesões")
            content = content.replace("S?bado", "Sábado")
            content = content.replace("Lan?ar", "Lançar")

            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            
            print(f"Fixed {filepath}")
        except Exception as e:
            print(f"Error on {filepath}: {e}")
